unixfs/archive/archive.go

```diff
@@ -29,15 +41,16 @@ func DagArchive(ctx cxt.Context, nd *mdag.Node, name string, dag mdag.DAGService
 	bufw := bufio.NewWriterSize(pipew, DefaultBufSize)
 
 	// compression determines whether to use gzip compression.
-	var maybeGzw io.Writer
+	var maybeGzw io.WriteCloser
+	var err error
```

Comment:
i thought this function (`DagArchive`) had gotten decomposed. maybe i misremember. it's really big and scary though...